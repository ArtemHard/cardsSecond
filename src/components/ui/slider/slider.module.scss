.SliderWrapper {
  display: flex;
  flex-direction: row;
  box-sizing: border-box;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  gap: 0.75rem;
}
.cardNumber {
    height: 2.25rem;
    width: 2.25rem;
    display: flex;
    justify-content: center;
    padding: 0.37rem 0.56rem;
    border: 1px solid var(--color-dark-300);
    border-radius: 0.125rem;
}
.SliderRoot {
  position: relative;
  display: flex;
  align-items: center;
  user-select: none;
  touch-action: none;
  width: 100%;
  height: 20px;
}

.SliderTrack {
  background-color: var(--color-accent-900);
  position: relative;
  flex-grow: 1;
  border-radius: 9999px;
  height: 0.35rem;
}

.SliderRange {
  position: absolute;
  background-color: var(--color-accent-500);
  border-radius: 9999px;
  height: 100%;
}

.SliderThumb {
  display: block;
  width: 12px;
  height: 12px;
  background-color: white;
  box-shadow: 0 0 0 6px var(--color-accent-500);
  border-radius: 10px;
  cursor: pointer;
  transition: all 0.4s ease-in-out;
}
.SliderThumb:hover {
  background-color: var(--color-accent-500);
  box-shadow: 0 0 0 6px white;
}
.SliderThumb:focus {
  outline: none;
}
