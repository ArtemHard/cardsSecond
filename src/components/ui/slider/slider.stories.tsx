import { useState } from 'react'

import { Meta, StoryObj } from '@storybook/react'

import { Slider } from '.'

const meta = {
  title: 'Components/Slider',
  component: Slider,
  // tags: ['autodocs'],
  // args: {
  //   value: [0, 100],
  // },
} satisfies Meta<typeof Slider>

export default meta
// type Story = StoryObj<typeof meta>

export const InComponent = {
  render: () => {
    const [value, setValue] = useState<number[]>([0, 75])
    const max = 75
    const min = 0

    const onValueCommit = (values: number[]) => {
      alert('установленно значние' + values)
    }

    console.log(value)

    return (
      <Slider
        onValueChange={setValue}
        min={min}
        max={max}
        value={value}
        onValueCommit={onValueCommit}
      />
    )
  },
}
