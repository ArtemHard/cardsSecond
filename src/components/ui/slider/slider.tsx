import { forwardRef } from 'react'

import * as RadixSlider from '@radix-ui/react-slider'

import { Typography } from '../Typography'

import style from './slider.module.scss'

export const Slider = forwardRef<HTMLSpanElement, RadixSlider.SliderProps>((props, ref) => {
  return (
    <div className={style.SliderWrapper}>
      <div className={style.cardNumber}>
        <Typography
          aria-label="minValue"
          variant="body1"
          as="span"
          style={{ position: 'absolute' }}
        >
          {props?.value?.[0]}
        </Typography>
      </div>
      <RadixSlider.Root
        defaultValue={props.value}
        minStepsBetweenThumbs={1}
        className={style.SliderRoot}
        onValueChange={props.onValueChange}
        onValueCommit={props.onValueCommit}
        max={props.max ?? 100}
        min={props.min ?? 0}
        ref={ref}
      >
        <RadixSlider.Track className={style.SliderTrack}>
          <RadixSlider.Range className={style.SliderRange} />
        </RadixSlider.Track>
        <RadixSlider.Thumb key={1} className={style.SliderThumb} />
        <RadixSlider.Thumb key={2} className={style.SliderThumb} />
      </RadixSlider.Root>
      <div className={style.cardNumber}>
        <Typography
          aria-label="maxValue"
          variant="body1"
          as="span"
          style={{ position: 'absolute' }}
        >
          {props?.value?.[1]}
        </Typography>
      </div>
    </div>
  )
})
