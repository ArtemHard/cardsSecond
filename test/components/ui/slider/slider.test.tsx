// eslint-disable-next-line import/default
import React from 'react'

import { RenderResult, render, screen } from '@testing-library/react'
import { afterEach, beforeEach, expect, describe, test, vi, it } from 'vitest'

import { Slider } from '../../../../src/components/ui/slider/index'

describe('test of basic params', () => {
  const sliderId = 'ajrgkjadbqdegbkjdsfg'
  const defaultNumbers = [0, 12]
  let slider: HTMLDivElement

  beforeEach(() => {
    const ResizeObserverMock = vi.fn(() => ({
      observe: vi.fn(),
      unobserve: vi.fn(),
      disconnect: vi.fn(),
    }))

    vi.stubGlobal('ResizeObserver', ResizeObserverMock)
    render(<Slider data-testid={sliderId} defaultValue={defaultNumbers} value={defaultNumbers} />)
    slider = screen.getByTestId(sliderId)
  })

  afterEach(() => {
    vi.clearAllMocks()
  })

  test('check default value', () => {
    const minValueElement = screen.getByLabelText('minValue').innerHTML
    const maxValueElement = screen.getByLabelText('maxValue').innerHTML

    expect(minValueElement).toBe(defaultNumbers[0].toString())
    expect(maxValueElement).toBe(defaultNumbers[1].toString())
    expect(slider.innerHTML).toBeDefined()
  })

  it('snap shot mathes', () => {
    const wrapper = render(<Slider />)

    expect(wrapper).toMatchSnapshot()
  })
})
