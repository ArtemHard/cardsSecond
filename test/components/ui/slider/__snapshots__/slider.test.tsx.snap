// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test of basic params > snap shot mathes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="_SliderWrapper_9aa7bb"
      >
        <div
          class="_cardNumber_9aa7bb"
        >
          <span
            aria-label="minValue"
            class="_body1_c177e4"
            style="position: absolute;"
          >
            0
          </span>
        </div>
        <span
          aria-disabled="false"
          class="_SliderRoot_9aa7bb"
          data-orientation="horizontal"
          data-testid="ajrgkjadbqdegbkjdsfg"
          dir="ltr"
          style="--radix-slider-thumb-transform: translateX(-50%);"
        >
          <span
            class="_SliderTrack_9aa7bb"
            data-orientation="horizontal"
          >
            <span
              class="_SliderRange_9aa7bb"
              data-orientation="horizontal"
              style="left: 0%; right: 88%;"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
          >
            <span
              aria-label="Minimum"
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="0"
              class="_SliderThumb_9aa7bb"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(12% + 0px);"
          >
            <span
              aria-label="Maximum"
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="12"
              class="_SliderThumb_9aa7bb"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
        </span>
        <div
          class="_cardNumber_9aa7bb"
        >
          <span
            aria-label="maxValue"
            class="_body1_c177e4"
            style="position: absolute;"
          >
            12
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="_SliderWrapper_9aa7bb"
      >
        <div
          class="_cardNumber_9aa7bb"
        >
          <span
            aria-label="minValue"
            class="_body1_c177e4"
            style="position: absolute;"
          />
        </div>
        <span
          aria-disabled="false"
          class="_SliderRoot_9aa7bb"
          data-orientation="horizontal"
          dir="ltr"
          style="--radix-slider-thumb-transform: translateX(-50%);"
        >
          <span
            class="_SliderTrack_9aa7bb"
            data-orientation="horizontal"
          >
            <span
              class="_SliderRange_9aa7bb"
              data-orientation="horizontal"
              style="left: 0%; right: 100%;"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
          >
            <span
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="0"
              class="_SliderThumb_9aa7bb"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style=""
              tabindex="0"
            />
          </span>
          <span
            style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
          >
            <span
              aria-orientation="horizontal"
              aria-valuemax="100"
              aria-valuemin="0"
              class="_SliderThumb_9aa7bb"
              data-orientation="horizontal"
              data-radix-collection-item=""
              role="slider"
              style="display: none;"
              tabindex="0"
            />
          </span>
        </span>
        <div
          class="_cardNumber_9aa7bb"
        >
          <span
            aria-label="maxValue"
            class="_body1_c177e4"
            style="position: absolute;"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="_SliderWrapper_9aa7bb"
    >
      <div
        class="_cardNumber_9aa7bb"
      >
        <span
          aria-label="minValue"
          class="_body1_c177e4"
          style="position: absolute;"
        />
      </div>
      <span
        aria-disabled="false"
        class="_SliderRoot_9aa7bb"
        data-orientation="horizontal"
        dir="ltr"
        style="--radix-slider-thumb-transform: translateX(-50%);"
      >
        <span
          class="_SliderTrack_9aa7bb"
          data-orientation="horizontal"
        >
          <span
            class="_SliderRange_9aa7bb"
            data-orientation="horizontal"
            style="left: 0%; right: 100%;"
          />
        </span>
        <span
          style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
        >
          <span
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="0"
            class="_SliderThumb_9aa7bb"
            data-orientation="horizontal"
            data-radix-collection-item=""
            role="slider"
            style=""
            tabindex="0"
          />
        </span>
        <span
          style="transform: var(--radix-slider-thumb-transform); position: absolute; left: calc(0% + 0px);"
        >
          <span
            aria-orientation="horizontal"
            aria-valuemax="100"
            aria-valuemin="0"
            class="_SliderThumb_9aa7bb"
            data-orientation="horizontal"
            data-radix-collection-item=""
            role="slider"
            style="display: none;"
            tabindex="0"
          />
        </span>
      </span>
      <div
        class="_cardNumber_9aa7bb"
      >
        <span
          aria-label="maxValue"
          class="_body1_c177e4"
          style="position: absolute;"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
